galaxykubeman/templates/rbac.yaml

```diff
@@ -0,0 +1,30 @@
+kind: Role
+apiVersion: rbac.authorization.k8s.io/v1beta1
+metadata:
+  name: {{ .Release.Namespace }}-rbac
+  namespace: {{ .Release.Namespace }}
+rules:
+  - apiGroups: ["", "extensions", "apps"]
+    resources: ["*"]
+    verbs: ["*"]
+  - apiGroups: ["batch"]
+    resources:
+      - jobs
+      - cronjobs
+    verbs: ["*"]
+
+---
+
+kind: RoleBinding
+apiVersion: rbac.authorization.k8s.io/v1beta1
+metadata:
+  name: {{ .Release.Namespace }}-rbac-binding
+  namespace: {{ .Release.Namespace }}
+subjects:
+  - kind: ServiceAccount
+    name: {{ .Values.rbac.ksa }}
+    namespace: {{ .Release.Namespace }}
+roleRef:
+  apiGroup: rbac.authorization.k8s.io
+  kind: Role
+  name: {{ .Release.Namespace }}-rbac
```

Comment:
I'd suggest adding the name resolution to helpers:
https://github.com/galaxyproject/galaxykubeman-helm/blob/master/galaxykubeman/templates/_helpers.yml#L4

especially this one since it's passed in multiple places so if we ever decide to change the naming scheme (or even add something like an override in values) we can easily do it in one place and have it propagate to both places. Not strictly necessary just a best-practice suggestion.

```
{{- define "galaxykubeman.rbacName" -}}
{{- default (printf "%s-%s" .Release.Namespace "rbac") .Values.rbac.nameOverride | trunc 63 | trimSuffix "-" -}}
{{- end -}}
```
Something like this should do it I think (default to `[namespace]-rbac` unless a name is specified in Values)

Reply:
Just one question - why would we use `galaxykubeman.rbacName` if the value is `rbac.serviceAccount` that gets passed in? the name specified in Values is the `serviceAccount`, so I'm confused as to what exactly this is doing

Reply:
This would just defining the string for the name for the Role (which is also mentioned in the RoleBinding), not related to the service account. Perhaps calling it `galaxykubeman.roleName` would make it less ambiguous?
The idea behind using this is that if someone later wants for example to change from `{{ .Release.Namespace}}-rbac` to something else, they wouldn't have to track down where it's mentioned (in this case it's just two places in the same file so not too bad), but it also adds the ability to add things like the override easily. The idea is basically that this `"galaxykubeman.rbacName"` defined in helpers would render the actual name, and then it'd be included in the template as just `{{ include "galaxykubeman.rbacName" }}`. If later on we want to change the default to include release name not just namespace for example (imagine a scenario where we want to deploy two galaxies for the same user in the same namespace, say each galaxy attached to a different workspace down the line), we could very easily change it in the helpers and not touch the templates.
Again though, it's definitely not necessary and especially where it's just two mentions of it so far, it's not a big deal, but figured it'd be a good thing to add since it's fairly easy to change, and at the same time a good example about the helpers which can bd really helpful for strings that need to be computed at templating time and are used in multiple places.

Reply:
ah okay that makes sense. I agree, I think adding it now makes future changes easier. will update now!

Reply:
so I should be able to add 
`{{- define "galaxykubeman.roleName" -}}
{{- default (printf "%s-%s" .Release.Namespace "rbac") .Values.rbac.nameOverride | trunc 63 | trimSuffix "-" -}}
{{- end -}}`
as in the `galaxykubeman.roleName` part is arbitrary?

Reply:
Yes!

Reply:
awesome! I just pushed the change for this and the `{{ include "galaxykubeman.rbacName" . }}` in the template, lmk what you think :)